File: frontend/src/services/orderService.js
import api from './api';

const orderService = {
  // Place order
  placeOrder: async (orderData) => {
    const response = await api.post('/orders', orderData);
    // Return the whole data object to be safe, as some APIs don't wrap in .data
    return response.data;
  },

  // Get order by ID
  getOrderById: async (orderId) => {
    const response = await api.get(`/orders/${orderId}`);
    return response.data.data;
  },

  // Get my orders
  getMyOrders: async (page = 0, size = 10) => {
    const response = await api.get('/orders/my-orders', {
      params: { page, size }
    });
    return response.data.data;
  },

  // Cancel order
  cancelOrder: async (orderId) => {
    const response = await api.put(`/orders/${orderId}/cancel`);
    return response.data.data;
  },

  // Get shop orders (for sellers)
  getShopOrders: async (shopId, status = null, page = 0, size = 10) => {
    const response = await api.get(`/orders/shop/${shopId}`, {
      params: { status, page, size }
    });
    return response.data.data;
  },

  // Update order status (for sellers)
  updateOrderStatus: async (orderId, status) => {
    const response = await api.put(`/orders/${orderId}/status`, { status });
    return response.data.data;
  },

  // Get seller orders (all orders for current seller's shop)
  getSellerOrders: async (status = null, page = 0, size = 10) => {
    try {
      const response = await api.get('/orders/seller/orders', {
        params: { status, page, size }
      });
      return response.data;
    } catch (error) {
      console.error('Error fetching seller orders:', error);
      throw error;
    }
  },

  // Initiate payment
  createPayment: async (paymentMethod, orderCode) => {
    // Map internal payment method to API path
    const pathMap = {
      'VNPAY': 'vn_pay',
      'MOMO': 'momo'
    };

    const path = pathMap[paymentMethod] || 'vn_pay';
    const response = await api.post(`/payment/${path}`, null, {
      params: { orderCode }
    });
    return response.data; // Should return the payment URL string or an object containing it
  },
};

export default orderService;

File: backend/src/main/java/com/ecommerce/platform/business/entity/service/OrderService.java
package com.ecommerce.platform.service;

import com.ecommerce.platform.dto.request.PlaceOrderRequest;
import com.ecommerce.platform.dto.request.UpdateOrderStatusRequest;
import com.ecommerce.platform.dto.response.OrderResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import java.util.List;

public interface OrderService {
    
    // Customer - Place Order
    List<OrderResponse> placeOrder(Long customerId, PlaceOrderRequest request);
    
    // Customer - Track Order
    OrderResponse getOrderById(Long orderId, Long userId);
    
    Page<OrderResponse> getCustomerOrders(Long customerId, Pageable pageable);
    
    OrderResponse cancelOrder(Long orderId, Long customerId);
    
    // Seller - Manage Orders
    Page<OrderResponse> getShopOrders(Long shopId, String status, Pageable pageable);
    
    OrderResponse updateOrderStatus(Long orderId, Long sellerId, UpdateOrderStatusRequest request);
}
